File: kotlin/2114.kt

fun mostWordsFound(sentences: Array<String>): Int {
    
    var maxWords: Int = 0

    for(sentence in sentences) {
        var spaces = sentence.length - sentence.replace(" ", "").length

        if(spaces + 1 > maxWords){
            maxWords = spaces + 1
        }
    }

    return maxWords
}

fun main(){
    var wordArray = arrayOf<String>("alice and bob love leetcode", "i thinks so too", "this is great thanks very much")
    println(mostWordsFound(wordArray))

    wordArray = arrayOf<String>("please wait", "continue to fight", "continue to win")
    println(mostWordsFound(wordArray))
}

File: javascript/2114.js

var mostWordsFound = function(sentences) {
    return sentences.map((sentence) => {
        return sentence.split(" ").length
    }).sort((a,b) => a - b).pop()
}

const inputs = [
    [
        "alice and bob love leetcode", 
        "i think so too", 
        "this is great thanks very much"
    ], [
        "please wait", 
        "continue to fight", 
        "continue to win"
    ],[
        "brjmzkatznpzhomilyyhaerytedrrwrkpuujbncfncgs",
        "u yn mwbuqymvtq z etdw p nx awixyietwg a jhsklavdagvhxdy",
        "v moahq bfhrdhxobkyccxbwduwbgsgdbngviidlaj"
    ]
]

inputs.forEach(input => console.log(mostWordsFound(input)))